### PR TITLE
To get groups claim for Group Overage scenario for Web API

### PR DESCRIPTION
**Pull request by Shama-K**
Updated Web API code to get all the groups in case of groups overage.

### CODE REVIEW

**Review by derisen — COMMENTED** [review]
Works like a charm, thanks Shama! Could you also update the readme as well, adding instructions about 'client secret' and 'offline_access' scopes to the api registration part? Also updating the 'About the Code' section / api part would be great.

**Review by derisen — APPROVED** [review]
LGTM!

**Review by derisen — APPROVED** [review]
Just installed and tested again. LGTM!

**Comment by idg-sam** [comment]
Looks good to me!
1. Should we grant admin consent in the instructions?
2. We should change existing manual steps for groups emission to tell user exactly how to add it in IDT and AT
3. New manual step should also be added for modifying app-config.json to add group IDs of GroupAdmin and GroupMember to SPA
4. 3. New manual step should also be added for modifying appsettings.json to add group IDs of GroupAdmin and GroupMember to API

**Review by idg-sam — CHANGES_REQUESTED** [review]
Looks good to me!

- Should grant admin consent in the instructions
- In script-based configuration:
   - ~~We should change additional manual stepsfor groups emission to tell user exactly how to add it in IDT and AT~~ Remove the manual step for group claim emission in IDT and AT, since it is already in place 
   - New manual step should also be added for modifying app-config.json to add group IDs of GroupAdmin and GroupMember to SPA
   - New manual step should also be added for modifying appsettings.json to add group IDs of GroupAdmin and GroupMember to API

**Review by TiagoBrenck — APPROVED** [review]
I am approving, but with the same observations as Sam

**Comment by Shama-K** [comment]
> Looks good to me!
> 
> * Should grant admin consent in the instructions
> * In script-based configuration:
>   
>   * ~We should change additional manual stepsfor groups emission to tell user exactly how to add it in IDT and AT~ Remove the manual step for group claim emission in IDT and AT, since it is already in place
>   * New manual step should also be added for modifying app-config.json to add group IDs of GroupAdmin and GroupMember to SPA
>   * New manual step should also be added for modifying appsettings.json to add group IDs of GroupAdmin and GroupMember to API

Addressed the suggestions.

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

File: chapter2/README.md [hunk 1]
```diff
@@ -395,7 +398,113 @@ ngOnInit(): void {
...
+                {
+                    await ProcessUserGroupsForOverage(context);
+                }
+            }
+    ```
+
+    **ProcessClaimsForGroupsOverage** method uses `GraphServiceClient` to retrieve groups for the signed-in user from [/me/memberOf](https://docs.microsoft.com/graph/api/user-list-memberof) endpoint. All the group ids are stored in Session.
+
+    ```csharp
+                private static async Task ProcessUserGroupsForOverage(TokenValidatedContext context)
```
Comment (kalyankrishna1) [inline chapter2/README.md:443]: >ProcessUserGroupsForOverage [](start = 42, length = 27)

you dont have to print the whole method

Comment (Shama-K) [reply chapter2/README.md]: Updated the Readme files.
Thanks.

**Review by derisen — APPROVED** [review]
Great work Shama, thanks. Just make sure to pull in the latest changes to master before merging.

**Pull request merged**